### PR TITLE
[HttpKernel] Bump dependencies and apply upstream parameter types

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179
| License       | MIT
| Doc PR        | N/A

This PR bumps some dependencies of the HttpKernel component in order to apply parameter type declarations.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Tests red.

File: src/Symfony/Component/Config/FileLocator.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function __construct($paths = [])
     /**
      * {@inheritdoc}
      */
-    public function locate(string $name, string $currentPath = null, $first = true)
+    public function locate(string $name, string $currentPath = null, bool $first = true)
```
Comment (derrabus) [inline src/Symfony/Component/Config/FileLocator.php]: Fixed by #33154.

**Comment by derrabus** [comment]
Remaining failures are unrelated.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**